scripts/initialize.sh

```diff
@@ -107,18 +111,43 @@ if [ ! -z "$authorization_amount" ]; then
    authorization_amount_opt="--authorization ${authorization_amount}"
 fi
 
-cd $KEEP_BEACON_SOL_PATH
-    
 printf "${LOG_START}Setting up staking...${LOG_END}"
 
-stake="npx hardhat initialize \
-    --network $NETWORK \
-    --owner ${stake_owner} \
-    --provider ${staking_provider} \
-    --operator ${operator} \
-    --beneficiary ${beneficiary} \
-    --authorizer ${authorizer}"
+mint="npx hardhat initialize:mint 
+   --network $NETWORK \
+   --owner ${stake_owner}"
+stake="npx hardhat initialize:stake 
+   --network $NETWORK \
+   --owner ${stake_owner} \
+   --provider ${staking_provider} \
+   --beneficiary ${beneficiary} \
+   --authorizer ${authorizer}"
+authorize="npx hardhat initialize:authorize 
+   --network $NETWORK \
+   --owner ${stake_owner} \
+   --provider ${staking_provider} 
+   --authorizer ${authorizer}"
+register="npx hardhat initialize:register 
+   --network $NETWORK \
+   --provider ${staking_provider} \
+   --operator ${operator}"
+
+# go to beacon
+cd $KEEP_BEACON_SOL_PATH
+
+# 'eval' command is used because of the optional params that can be pased to the
+# Hardhat tasks
+eval ${mint} ${stake_amount_opt}
+eval ${stake} ${stake_amount_opt}
+eval ${authorize} ${authorization_amount_opt}
+eval ${register}
 
-eval ${stake} ${stake_amount_opt} ${authorization_amount_opt}
+if [ "$skip_ecdsa_initialization" != true ]; then
```

Comment:
Maybe we could drop it?

Reply:
[eac324b](https://github.com/keep-network/keep-core/pull/3096/commits/eac324b24071316543f33f2755297b5bc396d683)